test/components/selectionBoxLayerTests.ts

```diff
@@ -1,193 +1,244 @@
 ///<reference path="../testReference.ts" />
 
-describe("SelectionBoxLayer", () => {
-  it("boxVisible()", () => {
-    let svg = TestMethods.generateSVG();
-    let sbl = new Plottable.Components.SelectionBoxLayer();
-    sbl.renderTo(svg);
+describe("Interactive Components", () => {
+  describe("SelectionBoxLayer", () => {
+
+    describe("Basic Usage", () => {
+      let svgWidth = 500;
+      let svgHeight = 500;
+
+      let svg: d3.Selection<void>;
+      let sbl: Plottable.Components.SelectionBoxLayer;
+
+      beforeEach(() => {
+        svg = TestMethods.generateSVG(svgWidth, svgHeight);
+        sbl = new Plottable.Components.SelectionBoxLayer();
+      });
+
+      it("can get and set the boxVisible() property", () => {
+        assert.strictEqual(sbl.boxVisible(), false, "The box is not visible by default");
+
+        sbl.renderTo(svg);
+        assert.strictEqual(sbl.boxVisible(), false, "The box is not visible by default after rendering to svg");
+
+        assert.strictEqual(sbl.boxVisible(true), sbl, "Setting the boxVisible attribute returns the selection box layer");
+        assert.strictEqual(sbl.boxVisible(), true, "Setting the boxVisible to true attribute works");
+        sbl.boxVisible(false);
+        assert.strictEqual(sbl.boxVisible(), false, "Setting the boxVisible attribute to false works");
+
+        svg.remove();
+      });
```

Comment:
This crosses a very fine line of testing, but I have thinking about how to effectively test getters... only to end with the conclusion of not to test them directly (The best way to test them directly is to pull the corresponding variable backing the getter but then this leads into all sorts of trouble).

The best thing I can think of is to utilize the getters as our means of detecting correctness for the rest of the code (If we use a setter, then the getter should definitely reflect the set value) (If a state-modifying function is called, the getter should definitely reflect the modified state).

Sure we can use these unit tests to prove that the function exists, but I am unsure of its usefulness beyond that (Also this specific idea would be not useful for typescript consumers)